cmake/SetupCompilerOptions.cmake

```diff
@@ -175,20 +175,33 @@ if(ENABLE_FORTRAN AND BLT_FORTRAN_FLAGS)
      message(STATUS "Updated CMAKE_Fortran_FLAGS to \"${CMAKE_Fortran_FLAGS}\"")
 endif()
 
-################################################
-# Support extra flags for the CUDA compiler.
-# N.B. must be ; delimited.
-################################################
-if(ENABLE_CUDA AND BLT_CUDA_FLAGS)
-    set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS};${BLT_CUDA_FLAGS}")
-    # Replace spaces with semicolons in the CUDA flags
-    foreach(flag_var CUDA_NVCC_FLAGS)
-      if(${flag_var} MATCHES " ")
-        string(REGEX REPLACE " " ";" ${flag_var} "${${flag_var}}")
-      endif(${flag_var} MATCHES " ")
-    endforeach(flag_var)
+############################################################
+# Map Legacy FindCUDA variables to native cmake variables
+# Note - we are intentionally breaking the semicolon delimited 
+# list that FindCUDA demanded of CUDA_NVCC_FLAGS so users
+# are forced to clean up their host configs.
+############################################################
+if (ENABLE_CUDA)
+   enable_language(CUDA) 
+   # where to find nvcc
+   if(CUDA_BIN_PATH)
```

Comment:
Please remove support for as many FindCUDA variables as possible.  I would like this as close to normal CMake conventions as possible.  This reduces the amount of upkeep we need to support going forward.  This applies to CUDA_BIN_PATH, CUDA_HOST_COMPILER, and CUDA_NVCC_FLAGS.

Reply:
Regarding CUDA_SEPARABLE_COMPILATION - I think it's best practice to do this in general, I don't think there is harm to doing separable compilation but it's certainly a pitfall folks can encounter. Should I just do that by default for the sake of killing the CUDA_SEPARABLE_COMPILATION flag?